casper/src/main/scala/coop/rchain/casper/Validate.scala

```diff
@@ -361,6 +363,15 @@ object Validate {
     }
   }
 
+  def deployCount[F[_]: Applicative: Log](b: BlockMessage): F[Either[InvalidBlock, ValidBlock]] =
+    if (b.header.get.deployCount == b.body.get.deploys.length) {
+      Applicative[F].pure(Right(Valid))
+    } else {
+      for {
+        _ <- Log[F].warn(ignore(b, s"block deploy count does not match to the amount of deploys."))
+      } yield Left(InvalidUnslashableBlock)
```

Comment:
I think we should make this one slashable.

Reply:
A bit late but I think the block hash mismatch should be slashable too. Don't forget to modify Casper.scala to handle invalid block